src/compiler/utilities.ts

```diff
@@ -2600,7 +2600,9 @@ namespace ts {
             parent.kind === SyntaxKind.PropertyDeclaration ||
             parent.kind === SyntaxKind.ExpressionStatement && node.kind === SyntaxKind.PropertyAccessExpression ||
             getNestedModuleDeclaration(parent) ||
-            isBinaryExpression(node) && node.operatorToken.kind === SyntaxKind.EqualsToken) {
+            isBinaryExpression(node) && node.operatorToken.kind === SyntaxKind.EqualsToken ||
+            isReturnStatement(parent) && node === parent.expression && (isFunctionExpression(node) || isArrowFunction(node))

```

Comment:
I'm pretty sure this is OK:

```suggestion
            isReturnStatement(parent))
```

Since any node that starts a jsdoc walk will want to see all relevant jsdocs, and a jsdoc on a return statement will always (I think?) be relevant.